### DISCUSSION TITLE
Add `branches` option for `workflow_dispatch`

### DISCUSSION THREAD

**Discussion by DrJume** [community/community#75933] [Actions]
### Select Topic Area

Product Feedback

### Body

Currently, the `workflow_dispatch` is missing an option to restrict the branches/tags it can be used on. It would be great to have a similar option as seen in push/pull_request.

This would allow for a production deployment Action to be only manually run on the `main` branch. 

The only workaround right now is to add a manual check to cancel the run if a wrong branch has been selected.

**Comment by henrikhelmers** [comment]
I have a similar request that I think would fit nicely with this one.

If I have specified a single branch, so there are no choices to be made **do not show a popup**. Instead, make the "Run workflow" dropdown button be green and execute the workflow right away.

**Comment by CarlSargunar** [comment]
Yes please!!! This would be so useful to have

**Comment by NicolasAizier** [comment]
Same request
